File: Placement Sheet/Arrays/Intersection_Of_Two_Sorted_Arrays.java
import java.util.ArrayList;
import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;

public class Intersection_Of_Two_Sorted_Arrays {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        System.out.print("Enter the size of the first array: ");
        int size1 = input.nextInt();

        ArrayList<Integer> arr1 = new ArrayList<>();
        for (int i = 0; i < size1; i++) {
            System.out.printf("Enter %d number element of the array: ", i + 1);
            arr1.add(input.nextInt());
        }

        System.out.print("Enter the size of the second array: ");
        int size2 = input.nextInt();

        ArrayList<Integer> arr2 = new ArrayList<>();
        for (int i = 0; i < size2; i++) {
            System.out.printf("Enter %d number element of the array: ", i + 1);
            arr2.add(input.nextInt());
        }
        input.close();

        ArrayList<Integer> result = findIntersection(arr1, size1, arr2, size2);

        printArrayList(result);
    }

    private static void printArrayList(ArrayList<Integer> arr) {
        for (int i = 0; i < arr.size(); i++) {
            System.out.print(arr.get(i) + " ");
        }
    }

    private static ArrayList<Integer> findIntersection(ArrayList<Integer> arr1, int size1, ArrayList<Integer> arr2,
            int size2) {
        ArrayList<Integer> resultArr = new ArrayList<>();
        Map<Integer, Integer> map = new HashMap<>();
        if (arr1.size() == 0 || arr2.size() == 0) {
            return resultArr;
        }

        for (int i = 0; i < size2; i++) {
            map.put(arr2.get(i), map.getOrDefault(arr2.get(i), 0) + 1);
        }

        System.out.println(map);

        for (int i = 0; i < size1; i++) {
            int currVal = arr1.get(i);
            System.out.println(currVal);
            if (map.getOrDefault(currVal, 0) > 0) {
                resultArr.add(currVal);
                map.replace(currVal, map.get(currVal) - 1);
            }
        }

        return resultArr;
    }
}

File: Placement Sheet/Arrays/Initial Startups/Python/Intersection_Of_Two_Sorted_Arrays.py
def intersectionOfTwoSortedArray(arr1, arr2):
    resultArr = []
    map = {}

    # Check if either array is empty...
    if(len(arr1) == 0 or len(arr2) == 0):
        return resultArr
    
    # Count the frequency of elements in arr2...
    for element in arr2:
        map[element] = map.get(element, 0)+1
    
    print(map)

    # Find intersection elements...
    for element in arr1:
        print(element)
        if(map.get(element, 0) > 0):
            resultArr.append(element)
            map[element] = map[element] - 1
    
    return resultArr

arr1 = []
arr2 = []

size1 = (int)(input("Enter the size of the 1st array: "))

for i in range(0, size1):
    data = (int)(input("Enter the data: "))
    arr1.append(data)

size2 = (int)(input("Enter the size of the 2nd array: "))

for i in range(0, size2):
    data = (int)(input("Enter the data: "))
    arr2.append(data)

print("Array1: ", arr1)
print("Array2: ", arr2)

result = intersectionOfTwoSortedArray(arr1=arr1, arr2=arr2)
print("Result: ", result)